tests/unit/test_cli.py

```diff
@@ -66,3 +69,54 @@ def foo(
         app()
 
     some.assert_called_with("y", 100, 100.5, True, "default", "optional")
+
+
+def test_collection_commands_account(mocker):
+    some = mock.Mock()
+    app = App(inspect.getfile(App))
+    acc_client = create_autospec(AccountClient)
+    mocker.patch("databricks.sdk.AccountClient.__new__", mock.Mock(return_value=acc_client))
+
+    @app.command(is_unauthenticated=False, is_collection=True)
+    def foo(
+        name: str,
+        age: int,
+        salary: float,
+        is_customer: bool,
+        a: AccountClient,
+        collection_workspace_id: int = 1234,
+        address: str = "default",
+        optional_arg: str | None = None,
+    ):
+        """Some comment"""
+        some(name, age, salary, is_customer, collection_workspace_id, address, optional_arg, a)
+
+    with mock.patch.object(sys, "argv", [..., FOO_COMMAND]):
+        app()
+
+    some.assert_called_with("y", 100, 100.5, True, 1234, "default", "optional", acc_client)
+
+
+def test_collection_commands_workspace(mocker):
+    some = mock.Mock()
+    app = App(inspect.getfile(App))
+    ws = create_autospec(WorkspaceClient)
+    mocker.patch("databricks.sdk.WorkspaceClient.__new__", mock.Mock(return_value=ws))
+
+    @app.command(is_unauthenticated=False, is_collection=True)
```

Comment:
if `is_collection=True`, then blueprint would inject `a: AccountClient` instead of `WorkspaceClient`